packages/shared/lib/core/wallet/actions/requestPersistedAsset.ts

```diff
@@ -1,12 +1,19 @@
+import { getErc20TokenMetadata } from '@core/layer-2'
+import { isValidEthereumAddress } from '@core/utils/crypto/utils'
 import { OFFICIAL_TOKEN_IDS } from '../constants'
 import { NotVerifiedStatus, VerifiedStatus } from '../enums'
 import { buildPersistedAssetFromIrc30Metadata } from '../helpers'
-import { IPersistedAsset } from '../interfaces'
+import { IErc20Metadata, IIrc30Metadata, IPersistedAsset } from '../interfaces'
 import { AssetVerification } from '../types'
-import { getIrc30MetadataFromFoundryOutput } from '../utils/getIrc30MetadataFromFoundryOutput'
+import { getIrc30MetadataFromFoundryOutput } from '../utils'
 
-export async function requestPersistedAsset(tokenId: string): Promise<IPersistedAsset> {
-    const tokenMetadata = await getIrc30MetadataFromFoundryOutput(tokenId)
+export async function requestPersistedAsset(tokenId: string, chainId?: number): Promise<IPersistedAsset | undefined> {
+    let tokenMetadata: IIrc30Metadata | IErc20Metadata | undefined
```

Comment:
We should use the `TokenMetadata` type here rather than the union

Reply:
pushed

Reply:
I was thinking, do we want to include the base token metadata here then?

Reply:
Yes, we should

Reply:
but why though? i think we never need to fetch the metadata for the base token

Reply:
In this specific case the variable `tokenMetadata` will never be of type `TokenMetadata` as its set by the functions `getErc20TokenMetadata` & `getIrc30MetadataFromFoundryOutput`. Maybe later on we can make something more generic, but for now I would stick to the strict types that we use